src/AutoRest.CSharp/Common/Input/Utf8JsonReaderExtensions.cs

```diff
@@ -129,7 +164,7 @@ public static bool TryReadWithConverter<T>(this ref Utf8JsonReader reader, strin
             return true;
         }
 
-        public static bool TryReadWithConverter<T>(this ref Utf8JsonReader reader, string propertyName, JsonSerializerOptions options, ref IReadOnlyList<T>? value)
+        public static bool TryReadComplexType<T>(this ref Utf8JsonReader reader, string propertyName, JsonSerializerOptions options, ref IReadOnlyList<T>? value)
```

Comment:
how about `TryReadComplextArrayType`?

Reply:
MTG has it in the same way: https://github.com/microsoft/typespec/blob/586a120eb425cf8aeb383ba76becc8a8fd6efa1d/packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator.Input/src/InputTypes/Serialization/Utf8JsonReaderExtensions.cs#L145
I would say arrays are also complex type, the overload actually should help us to make the code simpler because they just accept the same parameter list.

Reply:
well, I think it's kind of confusing for user to choose which method to use based on the ref parameter.
Anyway, this is just a copy from MTG.